src/patch/JsonPointer.ts

```diff
@@ -0,0 +1,42 @@
+interface JsonPointer {
+	segments(): string[];
+	toString(): string;
+	push(segment: String): JsonPointer;
+	pop(): JsonPointer;
+}
+export default JsonPointer;
+
+export function navigate(path: JsonPointer, target: any) {
+	return path.segments().reduce(function(prev: any, next: string) {
```

Comment:
Point 5 under the general coding [style guidelines](https://github.com/dojo/meta/blob/master/STYLE.md#general) indicates that arrow functions should *only* be used to preserve execution context. I take it that that is not correct?